lib/resque/job.rb

```diff
@@ -131,7 +140,13 @@ def self.destroy(queue, klass, *args)
           end
         end
       else
-        destroyed += data_store.remove_from_queue(queue, encode(:class => klass, :args => args))
+        cleaned_args = decode(encode(args))
+        data_store.everything_in_queue(queue).each do |string|
+          payload = decode(string)
+          if payload['class'] == klass && payload['args'] == cleaned_args
```

Comment:
NOTE: this is kinda nasty, but it needs to match any job with those arguments regardless of what's in the payload's metadata. To my knowledge, this function isn't called anywhere in `github/github` (since it wouldn't work)